### PR TITLE
Update module documentation - Part 2

### PR DESCRIPTION
**Pull request by SergioGasquez**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
Follow-up PR of https://github.com/esp-rs/esp-hal/pull/1763

#### Testing
Built the documentation and ran the doc-tests.

### CODE REVIEW

File: esp-hal/src/otg_fs.rs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+//! # USB On-The-Go (USB OTG)
 //!
 //! ## Overview
-//! The USB OTG Full-speed peripheral driver provides support for the USB
-//! On-The-Go (OTG) full-speed functionality on ESP chips, allows communication
+//! The USB OTG Full-speed (FS) peripheral allows communication
 //! with USB devices using either blocking (usb-device) or asynchronous
 //! (embassy-usb) APIs.
 //!
+//! It can operate as either a USB Host or Device, and supports fusll-speed (FS)
```
Comment (playfulFence) [inline esp-hal/src/otg_fs.rs:8]: Typo here: "fusll-speed" -> "full speed"

File: esp-hal/src/otg_fs.rs [hunk 2]
```diff
@@ -23,6 +26,15 @@
...
 //! `embassy-usb` crate.
+//!
+//! ## Examples
+//! Visit the [USB Serial] example for an example of using the USB OTG
+//! peripheral.
+//!
+//! [USB Serial]: https://github.com/esp-rs/esp-hal/blob/main/examples/src/bin/usb_serial.rs
+//!
+//! ## Implementation State
+//! - Low-speed (LS) not supported.
```
Comment (playfulFence) [inline esp-hal/src/otg_fs.rs:37]: "LS not supported" ->  "LS is not supported"

File: esp-hal/src/hmac.rs [hunk 3]
```diff
@@ -23,16 +16,23 @@
...
 //!
-//! The accelerator is available on ESP32-S2, ESP32-S3, ESP32-C3 and ESP32-C6.
+//! The HMAC padding is handled by the driver. In downstream mode, users do not
+//! need to input any message or apply padding. The HMAC module uses a default
+//! 32-byte pattern of 0x00 for re-enabling JTAG and a 32-byte pattern of 0xff
+//! for deriving the AES key for the DS module.
 //!
-//! # HMAC padding
+//! ## Examples
+//! Visit the [HMAC] example for an example of using the HMAC accelerator.
```
Comment (playfulFence) [inline esp-hal/src/hmac.rs:33]: Maybe we should avoid using two "example" words in a row? (just a nitpick)

Something like: "Visit the [HMAC] example to learn how to use the HMAC accelerator"

File: esp-hal/src/parl_io.rs [hunk 4]
```diff
@@ -1,112 +1,27 @@
...
-//! ```rust, no_run
-#![doc = crate::before_snippet!()]
-//! # use esp_hal::gpio::Io;
-//! # use esp_hal::dma_buffers;
-//! # use esp_hal::dma::{Dma, DmaPriority};
-//! # use esp_hal::parl_io::{no_clk_pin, BitPackOrder, ParlIoRxOnly, RxFourBits};
-//! # use crate::esp_hal::prelude::_fugit_RateExtU32;
-//!
-//! # let io = Io::new(peripherals.GPIO, peripherals.IO_MUX);
+//! See the [Parallel IO RX] example for an example of initializing the RX and
```
Comment (playfulFence) [inline esp-hal/src/parl_io.rs:15]: Same here ("example")

File: esp-hal/src/parl_io.rs [hunk 5]
```diff
@@ -1,112 +1,27 @@
...
-//!     &clocks,
-//! )
-//! .unwrap();
-//!
-//! let mut parl_io_rx = parl_io
-//!     .rx
-//!     .with_config(&mut rx_pins, no_clk_pin(), BitPackOrder::Msb, Some(0xfff))
-//!     .unwrap();
+//! ### Initialization for TX
+//! See the [Parallel IO TX] example for an example of initializing the TX and
```
Comment (playfulFence) [inline esp-hal/src/parl_io.rs:21]: Same here ("example")

File: esp-hal/src/reset.rs [hunk 6]
```diff
@@ -1,32 +1,19 @@
 //! # Hardware and Software Reset
 //!
 //! ## Overview
-//! The Hardware and Software Reset module provides functions for performing
-//! hardware and software resets on ESP chips. It also includes functions for
+//! Espressif chips provide four types of reset that occur at different levels,
+//! namely CPU Reset, Core Reset, System Reset, and Chip Reset. All reset types
```
Comment (playfulFence) [inline esp-hal/src/reset.rs:5]: Just a suggestion, feel free to decline: maybe let's put those mods to ``?
(Like `CPU Reset`)

File: esp-hal/src/rng.rs [hunk 7]
```diff
@@ -41,6 +26,20 @@
...
 #![doc = concat!("[ESP-IDF documentation](https://docs.espressif.com/projects/esp-idf/en/latest/", crate::soc::chip!(), "/api-reference/system/random.html)")]
+//! ## Configuration
+//! To use the [Rng] Driver, you need to initialize it with the RNG peripheral.
+//! Once initialized, you can generate random numbers by calling the `random`
+//! method, which returns a 32-bit unsigned integer.
+//!
+//! ## Usage
+//! This driver implements the [Read](embedded_hal_02::blocking::rng::Read)
+//! trait from the `embedded_hal` crate, allowing you to generate random bytes
+//! by calling the `read` method.
```
Comment (playfulFence) [inline esp-hal/src/rng.rs:37]: Again, just a suggestion. We can also remind that it implements traits from [rand_core](https://docs.rs/rand_core/0.6.4/rand_core/index.html)

**Review by playfulFence — CHANGES_REQUESTED** [review]
Thanks a lot for taking care of this! Here are some nitpicking in comments, some of them are more like suggestions.

**Review by JurajSadel — COMMENTED** [review]
I agree with @playfulFence's suggestions, otherwise LGTM, thanks for working on this!

File: esp-hal/src/i2s.rs [hunk 8]
```diff
@@ -73,6 +75,9 @@
 //! }
 //! # }
 //! ```
+//! 
+//! ## Implementation State
+//! - Only master mode is supported.
```
Comment (jessebraham) [inline esp-hal/src/i2s.rs:80]: IIRC we also only implement the TDM Phillips audio standard, which should probably be noted here as well

**Review by jessebraham — APPROVED** [review]
One small nitpick but otherwise I think it looks good, thanks!

**Review by MabezDev — APPROVED** [review]
Thanks!

**Comment by playfulFence** [comment]
Thanks a lot for all the changes, LGTM! 
Also sorry, I was AFK for a lunch, so I couldn't approve the changes! 😅

**Pull request merged**